week1/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM python:3.8-slim-buster
+RUN useradd --create-home appuser
+USER appuser
+
+# set a directory for the app
+WORKDIR /usr/src/app
+
+# copy all the files to the container
+COPY flask-app .
+
+# install dependencies
+RUN pip install -r requirements.txt --user
```

Comment:
let's install des before coping all file to utilize cache: 

```
COPY flask-app/requirements.txt requirements.txt
RUN pip install -r requirements.txt --user
COPY flask-app .
```

https://stackoverflow.com/questions/34398632/docker-how-to-run-pip-requirements-txt-only-if-there-was-a-change